sklearn/cluster/tests/test_k_means.py

```diff
@@ -974,3 +995,102 @@ def test_result_of_kmeans_equal_in_diff_n_jobs():
+
+    with pytest.warns(DeprecationWarning, match=depr_msg):
+        kmeans.fit(X)
+
+
+def test_warning_elkan_1_cluster():
+    X, _ = make_blobs(n_samples=10, n_features=2, centers=1, random_state=0)
+    kmeans = KMeans(n_clusters=1, n_init=1, init='random', random_state=0,
+                    algorithm='elkan')
+
+    with pytest.warns(RuntimeWarning,
+                      match="algorithm='elkan' doesn't make sense for a single"
+                            " cluster"):
+        kmeans.fit(X)
+
+
+def test_error_wrong_algorithm():
+    X, _ = make_blobs(n_samples=10, n_features=2, centers=2, random_state=0)
+    kmeans = KMeans(n_clusters=2, n_init=1, init='random', random_state=0,
+                    algorithm='wrong')
+
+    with pytest.raises(ValueError,
+                       match="Algorithm must be 'auto', 'full' or 'elkan'"):
+        kmeans.fit(X)
+
+
+@pytest.mark.parametrize("representation", ["dense", "sparse"])
+def test_relocate_empty_clusters(representation):
+    # test for the _relocate_empty_clusters_dense helper
+
+    # Synthetic dataset with 3 obvious clusters of different sizes
+    X = np.array(
+        [-10., -9.5, -9, -8.5, -8, -1, 1, 9, 9.5, 10]).reshape(-1, 1)
+    if representation == "sparse":
+        X = sp.csr_matrix(X)
+    sample_weight = np.full(10, 1.)
+
+    # centers all initialized to the first point of X
+    centers_old = np.array([-10., -10, -10]).reshape(-1, 1)
+
+    # With this initialization, all points will be assigned to the first center
+    # At this point centers_new is the weighted sum of the points it contains,
+    # if it's not empty, otherwise it the same as before.
+    centers_new = np.array([-16.5, -10, -10]).reshape(-1, 1)
+    weight_in_clusters = np.array([10., 0, 0])
+    labels = np.zeros(10, dtype=np.int32)
+
+    if representation == "dense":
+        _relocate_empty_clusters_dense(X, sample_weight, centers_old,
+                                       centers_new, weight_in_clusters, labels)
+    else:
+        _relocate_empty_clusters_sparse(X.data, X.indices, X.indptr,
+                                        sample_weight, centers_old,
+                                        centers_new, weight_in_clusters,
+                                        labels)
+
+    # The relocation scheme will take the 2 points farthest from the center and
+    # assign them to the 2 empty clusters, i.e. points at 10 and at 9.9. The
+    # first center will be updated to contain the other 8 points.
+    assert_array_equal(weight_in_clusters, [8, 1, 1])
+    assert_allclose(centers_new, [[-36], [10], [9.5]])
+
+
+@pytest.mark.parametrize("array_constr",
+                         [np.array, sp.csr_matrix],
+                         ids=['dense', 'sparse'])
+@pytest.mark.parametrize("algo", ['full', 'elkan'])
+def test_k_means_1_iteration(array_constr, algo):
+    # check k_means results for a single iteration (EME) vs pure python implem.
+    X = np.random.random_sample((100, 5))
```

Comment:
Please fix the RNG seed (to make it easier to reproduce the failure in case of regression).

Also I find the following more explicit:

```suggestion
    X = np.random.uniform(size=(100, 5))
```